### PR TITLE
Feat: redocly.yaml config structure

### PR DESCRIPTION
**Pull request by tatomyr**
## What/Why/How?

Pitch: [.redocly.yaml structure](https://intranet.redoc.ly/roadmap/tooling/2022-02/redocly-yaml/)

- [x]  new config format
- [x] push command new syntax

## Check yourself

- [x] Code is linted
- [x] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: __tests__/lint/info-contact--deprecated-format/snapshot.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`E2E lint info-contact--deprecated-format 1`] = `
+The apiDefinitions field is deprecated. Use apis instead.
```
Comment (RomanHotsiy) [inline __tests__/lint/info-contact--deprecated-format/snapshot.js:4]: We need to hide it until we publish docs
```suggestion
The apiDefinitions field is deprecated. Use apis instead, see changelog: <link here>
```

File: packages/cli/src/commands/push.ts [hunk 2]
```diff
@@ -41,92 +46,123 @@ export async function handlePush(argv: {
...
-        version,
-        filesHash,
-        filename: file.keyOnS3,
-        isUpsert: upsert,
-      });
-
-      if (file.filePath === filesToUpload.root) {
-        rootFilePath = filePath;
-      }
+  const apis = entrypoint ? { [`${name}@${version}`]: { root: entrypoint } } : config.apis;
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/push.ts:81]:
```suggestion
  // move here
  const apis = entrypoint ? { [`${name}@${version}`]: { root: getFallbackEntryPointsOrExit([entrypoint], config)[0] ) } } : config.apis;
```

File: packages/cli/src/commands/push.ts [hunk 3]
```diff
@@ -232,12 +266,73 @@ function hashFiles(filePaths: { filePath: string }[]) {
...
+  }: BarePushArgs) => {
+    if (!!maybeBranchName) {
+      process.stdout.write(
+        yellow(
+          'Deprecation warning: Do not use the third parameter as a branch name. Please use a separate --branch option instead.',
+        ),
+      );
+    }
+    const entrypoint =
+      maybeEntrypointOrDestination && fs.existsSync(maybeEntrypointOrDestination)
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/push.ts:313]: `opeapi push alias @org/api-name@api-version ...`

File: packages/cli/src/commands/push.ts [hunk 4]
```diff
@@ -232,12 +266,73 @@ function hashFiles(filePaths: { filePath: string }[]) {
...
+export function getApiEntrypoint({
+  name,
+  version,
+  config: { apis },
+}: {
+  name: string;
+  version: string;
+  config: Config;
+}): string {
+  const api = apis?.[`${name}@${version}`] || (version === DEFAULT_VERSION && apis?.[name]);
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/push.ts:334]: Check here too...

**Comment by tatomyr** [comment]
@RomanHotsiy I've resolved the comments, please see the last 2 commits.

File: packages/cli/src/commands/preview-docs/index.ts [hunk 5]
```diff
@@ -136,7 +136,7 @@ export async function previewDocs(argv: {
 
     const redoclyClient = new RedoclyClient();
     isAuthorizedWithRedocly = await redoclyClient.isAuthorizedWithRedocly();
-    const referenceDocs = config.referenceDocs || {};
+    const referenceDocs = config['features.openapi'] || {};
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/preview-docs/index.ts:139]: Can we use merged config here so `features.openapi` is used from the specific API ?

File: packages/cli/src/commands/push.ts (hunk 2 above)
Comment (tatomyr) [reply packages/cli/src/commands/push.ts:81]: Checked. This works correctly without changes.

File: packages/cli/src/commands/push.ts (hunk 4 above)
Comment (tatomyr) [reply packages/cli/src/commands/push.ts:331]: Checked. It works.

File: packages/cli/src/commands/preview-docs/index.ts (hunk 5 above)
Comment (tatomyr) [reply packages/cli/src/commands/preview-docs/index.ts]: Updated the code. Please see the commit: https://github.com/Redocly/openapi-cli/pull/588/commits/d95e9d89bb8d72cab75eb5804fdfe96666ccb20e

**Review by RomanHotsiy — APPROVED** [review]

**Pull request merged**